### PR TITLE
Post edit/delete

### PR DESCRIPTION
**Pull request by tanishk15sharma**
- [x] Style: post card options bar
- [x] Style: create post modal
- [x] Fix: Profile edit 
- [x] Fix: repeating code 
- [x] Feat : edit state `useState`
- [x] Feat : edit post
- [x] Feat : delete post from `db`
- [x] Feat : remove post from frontend

### CODE REVIEW

File: src/features/postSlice.js [hunk 1]
```diff
@@ -38,6 +41,7 @@ const postsSlice = createSlice({
     },
   },
 });
-export const { addPosts } = postsSlice.actions;
+console.log(postsSlice.actions);
```
Comment (ashishnnnnn) [inline src/features/postSlice.js:44]: Console can be removed if not required..

File: src/utils/posts.js [hunk 2]
```diff
@@ -118,11 +118,49 @@ const addComment = async (data, postId) => {
...
+    const res = await axios.put(
+      `${process.env.REACT_APP_BACKEND_URL}/posts/${postId}`,
+      { desc },
+      {
+        headers: {
+          token,
+        },
+      }
+    );
+    console.log(res);
```
Comment (ashishnnnnn) [inline src/utils/posts.js:133]: Same here.. can be removed

**Review by ashishnnnnn — APPROVED** [review]
Code seems perfectly fine to merge..
Some console log statement could be removed..
Liked that you used redux

File: src/components/CreatePost.js [hunk 3]
```diff
@@ -14,14 +14,18 @@ const CreatePost = () => {
     if (!desc.length) {
       return alert("please write something");
     }
+    if (editDetails) {
+      return editPost(editDetails._id, desc);
+    }
     const imageUrl = await uploadImage(imageFile);
     const newPost = await createNewPost(desc, imageUrl);
     newPost.userId = user;
+    console.log(newPost);
```
Comment (nerdyspook) [inline src/components/CreatePost.js:23]: you can remove the log from here.

**Review by nerdyspook — APPROVED** [review]
Code structure looks good to me.

**Pull request merged**